diff --git a/node_modules/react-native-getui/android/build.gradle b/node_modules/react-native-getui/android/build.gradle
index 7cb66cf..86d81cf 100644
--- a/node_modules/react-native-getui/android/build.gradle
+++ b/node_modules/react-native-getui/android/build.gradle
@@ -68,11 +68,4 @@ dependencies {
 
     implementation 'com.getui:gtsdk:3.2.13.0'  //个推SDK
     implementation 'com.getui:gtc:3.1.12.0'    //个推核心组件
-
-    implementation 'com.getui.opt:hwp:3.1.1'   // 华为
-    implementation 'com.getui.opt:xmp:3.3.0'   // 小米
-    implementation 'com.assist-v3:oppo:3.2.0'  // oppo
-    implementation 'com.assist-v3:vivo:3.1.1'  // vivo
-    implementation 'com.getui.opt:mzp:3.2.2'   // 魅族
-    implementation 'com.getui.opt:ups:3.0.3'   // ups，ups目前支持坚果，索尼，海信手机
 }
diff --git a/node_modules/react-native-getui/android/src/main/java/com/getui/reactnativegetui/GetuiModule.java b/node_modules/react-native-getui/android/src/main/java/com/getui/reactnativegetui/GetuiModule.java
index 1b1df8c..9065358 100755
--- a/node_modules/react-native-getui/android/src/main/java/com/getui/reactnativegetui/GetuiModule.java
+++ b/node_modules/react-native-getui/android/src/main/java/com/getui/reactnativegetui/GetuiModule.java
@@ -2,7 +2,7 @@ package com.getui.reactnativegetui;
 
 import android.content.Context;
 
-import android.support.annotation.Nullable;
+import androidx.annotation.Nullable;
 
 import com.facebook.react.bridge.Arguments;
 import com.facebook.react.bridge.Callback;
diff --git a/node_modules/react-native-getui/android/src/main/java/com/getui/reactnativegetui/PushIntentService.java b/node_modules/react-native-getui/android/src/main/java/com/getui/reactnativegetui/PushIntentService.java
index 4f66448..ec32933 100644
--- a/node_modules/react-native-getui/android/src/main/java/com/getui/reactnativegetui/PushIntentService.java
+++ b/node_modules/react-native-getui/android/src/main/java/com/getui/reactnativegetui/PushIntentService.java
@@ -5,7 +5,7 @@ package com.getui.reactnativegetui;
  */
 
 import android.content.Context;
-import android.support.annotation.RequiresPermission;
+import androidx.annotation.RequiresPermission;
 
 import com.facebook.react.bridge.Arguments;
 import com.facebook.react.bridge.WritableMap;
